x-pack/plugins/security_solution/public/management/links.ts

```diff
@@ -232,23 +232,32 @@ export const getManagementFilteredLinks = async (
 ): Promise<LinkItem> => {
   try {
     const currentUserResponse = await plugins.security.authc.getCurrentUser();
-    const privileges = calculateEndpointAuthz(
-      licenseService,
-      plugins.fleet?.authz,
-      currentUserResponse.roles
-    );
-    if (!privileges.canAccessEndpointManagement) {
+    const { canAccessEndpointManagement, canIsolateHost, canAccessResponseActionsHistory } =
+      calculateEndpointAuthz(licenseService, plugins.fleet?.authz, currentUserResponse.roles);
+
+    if (!canAccessEndpointManagement) {
       return excludeLinks([
         SecurityPageName.hostIsolationExceptions,
         SecurityPageName.responseActionsHistory,
       ]);
     }
-    if (!privileges.canIsolateHost || !privileges.canAccessResponseActionsHistory) {
-      const hostIsolationExceptionsApiClientInstance = HostIsolationExceptionsApiClient.getInstance(
-        core.http
-      );
-      const summaryResponse = await hostIsolationExceptionsApiClientInstance.summary();
+    const hostIsolationExceptionsApiClientInstance = HostIsolationExceptionsApiClient.getInstance(
+      core.http
+    );
+    const summaryResponse = await hostIsolationExceptionsApiClientInstance.summary();
+
+    if (!canAccessResponseActionsHistory) {
+      // <= enterprise license
+      if (!canIsolateHost && !summaryResponse.total) {
+        return excludeLinks([
+          SecurityPageName.hostIsolationExceptions,
+          SecurityPageName.responseActionsHistory,
+        ]);
+      }
+      return excludeLinks([SecurityPageName.responseActionsHistory]);
```

Comment:
Not needed now, but wondering if would be better having an empty array and push elements on each conditions and do only one call for `excludeLinks` as this is growing more and more.
Something like:

```typescript
const linksToExclude = [];

if (conditionA) {
 linksToExclude.push('A');
}
if (conditionB) {
 linksToExclude.push('B');
}

return excludeLinks(linksToExclude);
```